services.go

```diff
@@ -286,9 +361,32 @@ func (s *ServicesService) GetSlackService(pid interface{}, options ...OptionFunc
 // GitLab API docs:
 // https://docs.gitlab.com/ce/api/services.html#edit-slack-service
 type SetSlackServiceOptions struct {
-	WebHook  *string `url:"webhook,omitempty" json:"webhook,omitempty" `
-	Username *string `url:"username,omitempty" json:"username,omitempty" `
-	Channel  *string `url:"channel,omitempty" json:"channel,omitempty"`
+	WebHook                   *string `json:"webhook,omitempty" `
+	Username                  *string `json:"username,omitempty" `
+	Channel                   *string `json:"channel,omitempty"`
+	NotifyOnlyBrokenPipelines *bool   `json:"notify_only_broken_pipelines,omitempty"`
+	NotifyOnlyDefaultBranch   *bool   `json:"notify_only_default_branch,omitempty"`
+	PushEvents                *bool   `json:"push_events,omitempty"`
+	PushChannel               *string `json:"push_channel,omitempty"`
+	IssuesEvents              *bool   `json:"issues_events,omitempty"`
+	IssueChannel              *string `json:"issue_channel,omitempty"`
+	ConfidentialIssuesEvents  *bool   `json:"confidential_issues_events,omitempty"`
+	ConfidentialIssueChannel  *string `json:"confidential_issue_channel,omitempty"`
+	MergeRequestsEvents       *bool   `json:"merge_requests_events,omitempty"`
+	MergeRequestChannel       *string `json:"merge_request_channel,omitempty"`
+	TagPushEvents             *bool   `json:"tag_push_events,omitempty"`
+	TagPushChannel            *string `json:"tag_push_channel,omitempty"`
+	NoteEvents                *bool   `json:"note_events,omitempty"`
+	NoteChannel               *string `json:"note_channel,omitempty"`
+	ConfidentialNoteEvents    *bool   `json:"confidential_note_events,omitempty"`
+	// TODO: Currently, GitLab ignores this option (not implemented yet?), so
+	// there is no way to set it. Uncomment when this is fixed.
+	// See: https://gitlab.com/gitlab-org/gitlab-ce/issues/49730
+	//ConfidentialNoteChannel   *string `json:"confidential_note_channel,omitempty"`
+	PipelineEvents            *bool   `json:"pipeline_events,omitempty"`
+	PipelineChannel           *string `json:"pipeline_channel,omitempty"`
+	WikiPageEvents            *bool   `json:"wiki_page_events,omitempty"`
+	WikiPageChannel           *string `json:"wiki_page_channel,omitempty"`
```

Comment:
Please also add the `url` tags...

Reply:
I'm just curious why do we need this? As far as I understand, these fields are only encoded into JSON and send in a PUT request, but never in a query params.

Reply:
That is correct. So indeed for this specific options struct it would not be needed. But in order to keep things consistent and to prevent having to look at the type of HTTP call (e.g. `GET` or not) for every options struct, I prefer to just always add the `url` tags so however we use the struct, it will always work as expected.

Reply:
Done